### ISSUE TITLE
Incorrect SMBIOS tables

### ISSUE THREAD

**Issue by liuw** [cloud-hypervisor/cloud-hypervisor#4441]
**Describe the bug**

The end of table entry is not correct. Also there is an extraneous 0 after that entry.

**To Reproduce**

Launch a Linux guest and run `dmidecode`

**Version**

Main branch d9ee5c2640bb8eaaf1f3a687a8857fc5458401a7

** Output **

```
root@ubuntu:~# dmidecode
# dmidecode 3.2
Getting SMBIOS data from sysfs.
SMBIOS 3.2.0 present.
Table at 0x000F0018.

Handle 0x0001, DMI type 0, 20 bytes
BIOS Information
        Vendor: cloud-hypervisor
        Version: 0
        Release Date: Not Specified
        ROM Size: 64 kB
        Characteristics:
                PCI is supported
                System is a virtual machine

Handle 0x0002, DMI type 1, 27 bytes
System Information
        Manufacturer: Cloud Hypervisor
        Product Name: cloud-hypervisor
        Version: Not Specified
        Serial Number: Not Specified
        UUID: Not Settable
        Wake-up Type: Reserved
        SKU Number: Not Specified
        Family: Not Specified

Handle 0x0003, DMI type 127, 27 bytes
        <TRUNCATED>

Wrong DMI structures length: 130 bytes announced, structures occupy 131 bytes.
```

**Comment by liuw** [comment]
We should start by adding a correct `SystemEndOfTable` entry. Right now it is just using a wrong structure.

**Comment by liuw** [comment]
Oh well the MP table is also corrupting the SMBIOS table. Took me a while to figure that out...

**Comment by liuw** [comment]
Fixed by #4442
